In [1]:
import sklearn

In [2]:
from sklearn.datasets import load_breast_cancer

# Loading dataset
data = load_breast_cancer()

In [3]:
label_names = data['target_names']
labels = data['target']
feature_names = data['feature_names']
features = data['data']
#have lists for each set of information.

In [4]:
print(label_names)
print(labels[0])
print(feature_names[0])
print(features[0])

['malignant' 'benign']
0
mean radius
[1.799e+01 1.038e+01 1.228e+02 1.001e+03 1.184e-01 2.776e-01 3.001e-01
 1.471e-01 2.419e-01 7.871e-02 1.095e+00 9.053e-01 8.589e+00 1.534e+02
 6.399e-03 4.904e-02 5.373e-02 1.587e-02 3.003e-02 6.193e-03 2.538e+01
 1.733e+01 1.846e+02 2.019e+03 1.622e-01 6.656e-01 7.119e-01 2.654e-01
 4.601e-01 1.189e-01]


class names are malignant and benign we can represent them with binary values of 0 and 1.
0 represents malignant tumors 
1 represents benign tumors.

In [5]:
from sklearn.model_selection import train_test_split

In [6]:
#Split our data

train, test, train_labels, test_labels = train_test_split(features,labels,test_size=0.3,random_state=40)

test set that represents 30% of the original dataset. The remaining data (train) then makes up
the training data.

In [7]:
#naives bayes
from sklearn.naive_bayes import GaussianNB

In [8]:
#Initialize classifier
gnb = GaussianNB()

In [9]:
# Train classifier

model = gnb.fit(train, train_labels)

In [10]:
#predictions
preds = gnb.predict(test)
print(preds)

[1 0 1 1 0 1 1 1 0 1 0 1 1 1 0 1 1 1 1 1 1 0 1 0 0 0 0 0 1 1 1 0 0 1 1 1 1
 0 1 1 0 1 1 1 0 0 1 1 0 1 0 1 1 1 1 1 0 0 0 1 1 0 1 1 1 0 0 0 1 1 1 0 0 1
 0 0 0 1 1 1 1 1 0 1 0 1 0 0 1 0 1 1 1 1 1 0 1 1 1 1 1 1 1 0 1 1 1 0 1 1 1
 1 0 1 0 1 1 1 1 0 0 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 0 1 0 1 1 0 1 1 1 1
 1 0 1 0 1 0 1 1 0 1 0 1 0 1 1 1 1 1 1 1 0 1 0]


In [22]:
from sklearn.metrics import accuracy_score

# Evaluating accuracy
print("NB Accuracy:", accuracy_score(test_labels, preds))

NB Accuracy: 0.9649122807017544


NB classifier is 96% accurate. This means that 96% of the time the classifier is able to make the
correct prediction as to whether or not the tumor is malignant or benign.

## using SVM

In [12]:
from sklearn.svm import SVC

# Initialize SVM classifier
svm_classifier = SVC(kernel='linear', C=1)

# Train
svm_model = svm_classifier.fit(train, train_labels)

#predictions using SVM
svm_preds = svm_classifier.predict(test)
print(svm_preds)

# Evaluate accuracy
print("SVM Accuracy:", accuracy_score(test_labels, svm_preds))


[1 0 1 1 0 1 1 1 0 1 0 1 1 1 0 1 1 1 1 1 1 0 1 0 0 0 0 0 1 1 1 0 0 1 1 1 1
 0 1 1 0 1 1 1 0 0 1 0 0 1 0 1 1 1 1 1 0 0 0 1 1 1 1 1 1 0 0 0 1 1 1 1 0 0
 0 0 0 1 1 1 1 1 0 1 0 1 0 0 1 0 1 1 1 1 1 0 1 1 1 1 1 1 1 0 1 1 1 0 1 1 1
 1 0 1 0 1 1 1 1 0 0 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 0 1 0 1 1 0 1 1 1 1
 1 0 1 0 1 1 1 1 0 1 0 1 0 1 1 1 1 1 1 1 0 0 0]
SVM Accuracy: 0.9532163742690059


## using DT

In [13]:
from sklearn.tree import DecisionTreeClassifier

# Initialize the Decision Tree classifier
dt_classifier = DecisionTreeClassifier()

# Train the Decision Tree classifier
dt_model = dt_classifier.fit(train, train_labels)

# Make predictions using Decision Tree
dt_preds = dt_classifier.predict(test)
print(dt_preds)

# Evaluate accuracy for Decision Tree
print("Decision Tree Accuracy:", accuracy_score(test_labels, dt_preds))


[1 0 1 1 0 1 1 1 0 1 0 1 1 1 0 1 1 1 0 1 1 0 1 0 0 0 0 0 1 1 1 0 0 1 1 1 1
 0 1 1 0 1 1 1 0 0 1 1 0 1 0 1 1 1 1 1 0 1 0 1 1 0 1 1 1 0 0 0 1 1 0 0 0 1
 0 0 0 1 1 1 1 1 0 1 0 1 0 0 1 0 1 1 1 1 1 0 1 1 1 1 1 1 1 0 1 1 1 0 1 1 1
 1 0 1 0 1 1 1 1 0 0 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 1 1 1
 1 0 1 0 1 0 1 1 0 1 0 1 0 1 1 1 1 1 1 1 0 0 0]
Decision Tree Accuracy: 0.9590643274853801


In [14]:
from sklearn.ensemble import RandomForestClassifier

# Initialize the Random Forest classifier
rf_classifier = RandomForestClassifier(n_estimators=100, random_state=40)

# Train the Random Forest classifier
rf_model = rf_classifier.fit(train, train_labels)

# Make predictions using Random Forest
rf_preds = rf_classifier.predict(test)
print(rf_preds)

# Evaluate accuracy for Random Forest
print("Random Forest Accuracy:", accuracy_score(test_labels, rf_preds))


[1 0 1 1 0 1 1 1 0 1 0 1 1 1 0 1 1 1 1 1 1 0 1 0 0 0 0 0 1 1 1 0 0 1 1 1 1
 0 1 1 0 1 1 1 0 0 1 1 0 1 0 1 1 1 1 1 0 1 0 1 1 0 1 1 1 0 0 0 1 1 1 0 0 1
 0 0 0 1 1 1 1 1 0 1 0 1 0 0 1 0 1 1 1 1 1 0 0 1 1 1 1 1 1 0 1 1 1 0 1 1 1
 1 0 1 0 1 1 1 1 0 0 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 1 0 1
 1 0 1 0 1 0 1 1 0 1 0 1 0 1 1 1 1 1 1 1 0 0 0]
Random Forest Accuracy: 0.9707602339181286


In [20]:
from sklearn.linear_model import LogisticRegression

# Initialize the Logistic Regression classifier
logreg_classifier = LogisticRegression(max_iter=100)

# Train the Logistic Regression classifier
logreg_model = logreg_classifier.fit(train, train_labels)

# Make predictions using Logistic Regression
logreg_preds = logreg_classifier.predict(test)
print(logreg_preds)

# Evaluate accuracy for Logistic Regression
print("Logistic Regression Accuracy:", accuracy_score(test_labels, logreg_preds))


[1 0 1 1 0 1 1 1 0 1 0 1 1 1 0 1 1 1 1 1 1 0 1 0 0 0 0 0 1 1 1 0 0 1 1 0 1
 0 1 1 0 1 1 1 0 1 1 0 0 1 0 1 1 1 1 1 0 0 0 1 1 1 1 1 1 0 0 0 1 1 1 1 0 0
 0 0 0 1 1 1 1 1 0 1 0 1 0 0 1 0 1 1 1 1 1 0 1 1 1 1 1 1 1 0 1 1 1 0 1 1 1
 1 0 1 0 1 1 1 1 0 0 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 0 1 0 1 1 0 0 1 1 1
 1 0 1 0 1 1 1 1 0 1 0 1 0 1 1 1 1 1 1 1 0 0 0]
Logistic Regression Accuracy: 0.935672514619883


/Users/poojaharihar/anaconda3/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


# Classification Model Evaluation Report

## Random Forest Classifier
- **Accuracy:** 97.08%
- The Random Forest classifier achieved the highest accuracy among the models evaluated, making it a strong candidate for the breast cancer classification task.

## Decision Tree Classifier
- **Accuracy:** 95.91%
- The Decision Tree classifier performed well, demonstrating high accuracy in predicting breast cancer malignancy.

## Support Vector Machine (SVM) Classifier
- **Accuracy:** 95.32%
- The SVM classifier showed strong performance, although slightly lower than Random Forest. SVMs are known for their effectiveness in high-dimensional spaces.

## K-Nearest Neighbors (KNN) Classifier
- **Accuracy:** 93.57%
- The KNN classifier demonstrated good accuracy but performed slightly below some of the other models.

## Naive Bayes (NB) Classifier
- **Accuracy:** 96.49%
- The Naive Bayes classifier performed well, showcasing competitive accuracy in the breast cancer classification task.

---